zerver/lib/create_user.py

```diff
@@ -127,4 +132,21 @@ def create_user(email: str, password: Optional[str], realm: Realm,
     user_profile.save(update_fields=["recipient"])
 
     Subscription.objects.create(user_profile=user_profile, recipient=recipient)
+
+    if user_avatar_url:
+        from zerver.lib.upload import upload_avatar_image_from_url
+        from zerver.lib.actions import do_change_avatar_fields
+
+        try:
+            avatar_hash = upload_avatar_image_from_url(user_avatar_url, user_profile)
+            do_change_avatar_fields(user_profile, UserProfile.AVATAR_FROM_USER)
+            user_profile.avatar_hash = avatar_hash
+            user_profile.save(update_fields=["avatar_hash"])
+
+        except BadImageError as err:
```

Comment:
I don't think this is the only possible error? For example i thinkw e can get `requests.get` failures that will raise different exceptions? If we can't figure out the full set of reasonable exception to catch, I guess we should just catch everything :thinking:  But the logging message below should say something more than just str(exception) to make it clearer where this is happning

Reply:
`'Error while trying to set user avatar from URL %s: %s' % (user_avatar_url, err)` perhaps